test-sites/__fixtures__/README.md

```diff
@@ -0,0 +1,31 @@
+# Static API Mocks
+
+## Static API
+Our Hugo builds output json data files, and other builds consume these data
+files. For example:
+ - **Building ocw-www:**
+   - **produces** instructor data, e.g., https://ocw.mit.edu/instructors/a32373bc-8c78-e37a-f79e-1f15686d1a9b/index.json
+   - **consumes** course summary data (e.g., for the "Featured Courses" carousel)
+- **Building course sites:**
+  - **produces:** course summary data, e.g., https://ocw.mit.edu/courses/8-01sc-classical-mechanics-fall-2016/data.json
+  - **consumes:** instructor data (e.g., for the "Course Info" on course homepage)
+
+These data comprise our "static API". When Hugo consumes this data during the build, it does so via requests through `STATIC_API_BASE_URL`.
+
+## Circular dependencies when building a snapshot
+Because course builds depend on www data, and www builds depend on course data, it is possible (and likely) for builds to contain circular dependencies. For example, if
+ - Course A uses instructor X, and
+ - ocw-www markdown describes instructor X
+ - ocw-www features Course A as a featured course
+ - neither Course A nor ocw-www have ever been published
+then publishing will be deadlocked: the www build will fail for lack of course X, and the course X build will fail for lack of instructor data.
+
+Thus, it is likely that a **snapshot**[^1] of the OCW Markdown at a single point in time cannot be built in isolation by Hugo.
+
+[^1]: OCW Studio forces you to build the sites incrementally: Course A cannot be published until a WWW version has been published which includes the relevant instructors, and a course cannot be featured on WWW until it has been published.
+
+## Building a snapshot for e2e testing: `test-sites/__fixtures`
+
+For e2e testing we *do* want to build a snapshot of OCW markdown in isolation. But there's really only a circular dependency if the `STATIC_API_BASE_URL` is the Hugo output. So for e2e testing, we pre-construct the relevant JSON files (stored in `test-sites/__fixtures__`) and serve them for use during the Hugo builds.
```

Comment:
I have two overall thoughts here:
1. This circular dependency seems bad for OCW in general, not just the e2e tests. I understand that Studio mitigates it, but I suspect that if we created a new "staging" environment (as opposed to "rc"), and synced the database from prod -> staging, we'd have trouble publishing.  @gumaerc I think we did NOT run into this when you did the prod -> rc sync because rc was already "close enough" to prod. Does that sound right?

    (I'm not suggesting this is a big problem or something we should tackle right now.)
2. There are probably better ways to solve this issue for both the e2e tests AND "real" ocw. One idea would be:  Make the build tolerate missing data more consistently, then we could build twice: once would produce the json files (and not-quite-correct html), then the second pass would produce correct results for both.